cc-mt-service/src/routes/mtServiceRoutes.ts

```diff
@@ -0,0 +1,25 @@
+// Mt service route module
+import { Router, Request, Response } from "express";
+import {postAdvert, parseToken} from '../handleData';
+import {AdvertData, Error} from '../types';
+
+const mtRouter = Router();
+
+// Route for posting a new advert: /v1/advert
+mtRouter.post('/advert', async (req:Request, res:Response) => {
```

Comment:
So, this function needs some changes. First, check that there is token. For example, add check after the line 11, if wholeToken equals empty string. If yes, error response should be sent. Error message would be something like "No token available" and error status would be 401. Then, when you get the response from postAdvert, check if response.status is 401. That means it the token is invalid and you should set your own error message  again to "Invalid token" or something like that, and error status to 401.